### PR TITLE
Update to gulp 4.

### PR DESCRIPTION
**Pull request by kevincox**
This fixes the build for recent versions of node.

### CODE REVIEW

File: gulpfile.coffee [hunk 1]
```diff
@@ -122,30 +119,29 @@ gulp.task('install.rdf', ->
     .pipe(gulp.dest(DEST))
 )
 
-gulp.task('templates', [
+gulp.task('templates', gulp.series(
```
Comment (lydell) [inline gulpfile.coffee]: Should this be parallel?

File: gulpfile.coffee [hunk 2]
```diff
@@ -122,30 +119,29 @@ gulp.task('install.rdf', ->
...
+))
 
-gulp.task('push', ['xpi'], ->
+gulp.task('push', gulp.series('xpi', ->
   body = fs.readFileSync(join(DEST, XPI))
   request.post({url: 'http://localhost:8888', body})
-)
+))
+
+gulp.task('default', gulp.series(['push']))
```
Comment (lydell) [inline gulpfile.coffee]: Is `gulp.series` supposed to be given an array here? The other `gulp.series` are not given arrays.

File: gulpfile.coffee (hunk 1 above)
Comment (kevincox) [reply gulpfile.coffee]: It can be if it works. Changed.

File: gulpfile.coffee (hunk 2 above)
Comment (kevincox) [reply gulpfile.coffee]: My mistake. Fixed.

**Comment by lydell** [comment]
What does this fix, and for what version of Node?

**Comment by kevincox** [comment]
It fixes this issue: https://github.com/nodejs/node/issues/20285

All gulp commands were failing for me and I tried a number of things but upgrading gulp helped.

My node version is: 

```
% node --version
v10.0.0
```

**Comment by lilydjwg** [comment]
I tried to build VimFx today and gulp with node 10 failed with coredumps. I heard from others that node 10 has broken all their projects so I turned to an old node.

**Comment by kevincox** [comment]
@lilydjwg Does this patch fix it for you?

**Comment by lilydjwg** [comment]
@kevincox Yes, I just tried and it worked.

**Comment by lydell** [comment]
Thanks!

**Pull request merged**